### PR TITLE
Data Masking included in Message Analysis.md

### PR DESCRIPTION
**Pull request by medhaanand**

### CODE REVIEW

File: docs/bot-analytics/using-message-analysis.md [hunk 1]
```diff
@@ -182,3 +182,60 @@ Apart from the filter options, you also have a button to clear all filters you h
...
+
+## Data Masking
+
+Personal information in a conversation needs to be protected when a conversation is marked as complete. This includes User Says, Bot Says, Agent Messages, Media, etc. We mask or encrypt this personal information in order to protect it. Data masking is the practice of intentionally randomizing anonymizing specific data/information to secure it. 
+
+We mask the **original_text** not **entity_value**. In a few scenarios, the original_text and entity value are different. The current masking rules consider only the original_text. In the below image, it is shown how **original_text** and **entity_value** can be different -
+
+![PersonalEntities](assets/DataMasking3.png)
+
+Data on **Message Analysis** screen will show up in the masked format. If you download the **Message Analysis report**, then that will also show up with the masked version. However, chat links will show up in their original form. For easier bot maintenance, the masked value will also be tagged with what kind of information it is. Below image shows how a masked entity value will look like
```
Comment (medhaanand) [inline docs/bot-analytics/using-message-analysis.md:194]:
```suggestion
Data on **Message Analysis** screen will show up in the masked format. If you download the **Message Analysis report**, then that will also show up with the masked version. However, chat links will show up in their original form. 

The masked data is shown as 6 asterisks in a message as shown below -
```

File: docs/bot-analytics/using-message-analysis.md [hunk 2]
```diff
@@ -182,3 +182,61 @@ Apart from the filter options, you also have a button to clear all filters you h
...
+
+Data on **Message Analysis** screen will show up in the masked format. If you download the **Message Analysis report**, then that will also show up with the masked version. However, chat links will show up in their original form. 
+
+The masked data is shown as 6 asterisks in a message as shown below -
+
+![PersonalEntities](assets/DataMasking4.png)
+
+### What all gets masked? 
+
+Not everthing needs to be protected but sensitive data like phone-number, email-id, account number, date-of-birth, etc. needs to be protected. On the platform, these values are inserted as **entities**. So we essentially need to mask these sensitive entities.
```
Comment (medhaanand) [inline docs/bot-analytics/using-message-analysis.md:202]:
```suggestion
Not everything needs to be protected but sensitive data like phone-number, email-id, account number, date-of-birth, etc. needs to be protected. On the platform, these values are inserted as **entities**. So we essentially need to mask these sensitive entities.
```

File: docs/bot-analytics/using-message-analysis.md [hunk 3]
```diff
@@ -182,3 +182,61 @@ Apart from the filter options, you also have a button to clear all filters you h
...
+
+## Data Masking
+
+Personal information in a conversation needs to be protected when a conversation is marked as complete. This includes User Says, Bot Says, Agent Messages, Media, etc. We mask or encrypt this personal information in order to protect it. Data masking is the practice of intentionally randomizing anonymizing specific data/information to secure it. 
+
+We mask the **original_text** not **entity_value**. In a few scenarios, the original_text and entity value are different. The current masking rules consider only the original_text. In the below image, it is shown how **original_text** and **entity_value** can be different -
+
+![PersonalEntities](assets/DataMasking3.png)
+
+Data on **Message Analysis** screen will show up in the masked format. If you download the **Message Analysis report**, then that will also show up with the masked version. However, chat links will show up in their original form.
```
Comment (jatin3893) [inline docs/bot-analytics/using-message-analysis.md:194]: However, **the messages on** chat links will show up in their original form.

**chat links** should have a reference to chat link documentation page.

File: docs/bot-analytics/using-message-analysis.md [hunk 4]
```diff
@@ -182,3 +182,61 @@ Apart from the filter options, you also have a button to clear all filters you h
...
+
+The masked data is shown as 6 asterisks in a message as shown below -
+
+![PersonalEntities](assets/DataMasking4.png)
+
+### What all gets masked? 
+
+Not everything needs to be protected but sensitive data like phone-number, email-id, account number, date-of-birth, etc. needs to be protected. On the platform, these values are inserted as **entities**. So we essentially need to mask these sensitive entities.
+
+> Nothing else gets masked except **entities**.
```
Comment (jatin3893) [inline docs/bot-analytics/using-message-analysis.md:204]: Not sure if this phrase is valid.
Regex patterns will tag phrases which may not correspond to an entity.

File: docs/bot-analytics/using-message-analysis.md [hunk 5]
```diff
@@ -182,3 +182,61 @@ Apart from the filter options, you also have a button to clear all filters you h
...
+
+Data masking will be enabled by default for all new partners being created. If you wish to switch off the Data Masking (Data Encryption) flag, you can do the same from the Partners List under Admin Tools as shown below
+
+![PersonalEntities2](assets/DataMasking2.png)
+
+We do not recommend disabling Data Masking. If you do, then please explicitly inform the same among all stakeholders.
+
+### Can we see the unmasked/decrypted message?
+
+Even though it is technically possible to view the decrypted message, we do not allow the same anywhere. If absolutely required, then please get an approval from Haptik's Infosec team before using it in any application.
```
Comment (medhaanand) [inline docs/bot-analytics/using-message-analysis.md:242]:
```suggestion
```

File: docs/bot-analytics/using-message-analysis.md [hunk 6]
```diff
@@ -182,3 +182,59 @@ Apart from the filter options, you also have a button to clear all filters you h
...
+
+### Enable Data Masking
+
+Data masking will be enabled by default for all new partners being created. If you wish to switch off the Data Masking (Data Encryption) flag, you can do the same from the Partners List under Admin Tools as shown below
+
+![PersonalEntities2](assets/DataMasking2.png)
+
+We do not recommend disabling Data Masking. If you do, then please explicitly inform the same among all stakeholders.
+
+### Can we see the unmasked/decrypted message?
```
Comment (medhaanand) [inline docs/bot-analytics/using-message-analysis.md:240]:
```suggestion
```

File: docs/bot-analytics/using-message-analysis.md (hunk 3 above)
Comment (medhaanand) [reply docs/bot-analytics/using-message-analysis.md:194]: There is no such docs currently. Once I add, will add the link here too.

File: docs/bot-analytics/using-message-analysis.md (hunk 4 above)
Comment (medhaanand) [reply docs/bot-analytics/using-message-analysis.md:204]: Is it? I thought only entities are masked. Thats what we discussed yesterday right?

**Review by jatin3893 — APPROVED** [review]
LGTM

**Pull request merged**